src/main/java/com/myhome/services/springdatajpa/UserSDJpaService.java

```diff
@@ -62,17 +64,24 @@ public UserSDJpaService(UserRepository userRepository,
     return createUserInRepository(request);
   }
 
-  @Override public Set<User> listAll() {
+  @Override public List<User> listAll(String sort) {
     Set<User> userListSet = new HashSet<>();
     userRepository.findAll().forEach(userListSet::add);
-    return userListSet;
+
+    boolean isAscendingSort = sort.contentEquals("asc");
+    Comparator<User> comparator = isAscendingSort ? Comparator.comparing(User::getName)
```

Comment:
Sorting on code side is not a good practice.
It should be done on database side. `JpaRepository` have methods with argument `Sort sort` that can return database results sorted with the required order.
Take a look at: https://www.baeldung.com/spring-data-jpa-pagination-sorting

Reply:
Thanks for the comment. That was the exact reason why I marked this PR as a draft. I am working on implementing sorting, paging and filtering.